### ISSUE TITLE
TECH DEBT: Remove 'metadata' from data context and form

### ISSUE THREAD

**Issue by githubbob42** [githubbob42/mingle2github2#5773]
**[Mingle Card: 6139](https://liquidframeworks.mingle.thoughtworks.com/projects/alpine_mobile/cards/6139)**
Narrative
---------

The description is completely cloned and fields are cloned repeatedly for data context and form. Don’t do this.

Acceptance Criteria
-------------------

No regressions

_Analysis_
----------

Description cloned repeatedly to gain access to fields by name. Instead of mutating the description every time it’s used, save a \`fieldsByName\` property that provides \`fields\` information in a way that we need it throughout the codebase.

Related Stories
---------------

Tasks
-----

{{ table query: SELECT Number, Name, Owner, 'Task Status' WHERE Type = Task and Story = THIS CARD ORDER BY Status }}

Defects
-------

{{ table query: SELECT Number, Name, Owner, 'Status' WHERE Type = Defect and 'Related Story' = THIS CARD }}

Impact Analysis
---------------

_Basically everything_

Test Plan
---------

Full regression and iOS, but emphasis on:

1.  forms
2.  reports
3.  rule engine
4.  sync
5.  rollups
6.  ticket/quote items (create and copy)
7.  auto-select lookup filters
8.  field updates

**Comment by githubbob42** [comment]
Natasha : 12/17/2018 11:44 CST
since this card requires to test everything we've decided to merge it into hunterpeak and then all of qa's will perform a pre release full regression testing.

**Comment by githubbob42** [comment]
**Pull Request #3006**
